misc/compiler-plugins/modules/pkg-analyzer/src/main/java/org/ballerinalang/pkg/analyzer/PackageAnalyzerPlugin.java

```diff
@@ -0,0 +1,131 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.pkg.analyzer;
+
+import io.ballerina.projects.Module;
+import io.ballerina.projects.ModuleId;
+import io.ballerina.projects.Package;
+import io.ballerina.projects.ProjectException;
+import io.ballerina.projects.plugins.CodeAnalysisContext;
+import io.ballerina.projects.plugins.CodeAnalyzer;
+import io.ballerina.projects.plugins.CompilationAnalysisContext;
+import io.ballerina.projects.plugins.CompilerPlugin;
+import io.ballerina.projects.plugins.CompilerPluginContext;
+import io.ballerina.toml.semantic.TomlType;
+import io.ballerina.toml.semantic.ast.TomlArrayValueNode;
+import io.ballerina.toml.semantic.ast.TomlKeyValueNode;
+import io.ballerina.toml.semantic.ast.TomlStringValueNode;
+import io.ballerina.toml.semantic.ast.TomlTableNode;
+import io.ballerina.toml.semantic.ast.TomlValueNode;
+import io.ballerina.toml.semantic.ast.TopLevelNode;
+import io.ballerina.toml.semantic.diagnostics.TomlDiagnostic;
+import io.ballerina.toml.semantic.diagnostics.TomlNodeLocation;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@code PackageAnalyzerPlugin} adds diagnostics after compilation phase.
+ *
+ * @since 2.0.0
+ */
+public class PackageAnalyzerPlugin extends CompilerPlugin {
+    @Override
+    public void init(CompilerPluginContext pluginContext) {
+        pluginContext.addCodeAnalyzer(new PackageAnalyzer());
+    }
+
+    /**
+     * A sample {@code PackageAnalyzer} that adds diagnostics after compilation phase.
+     *
+     * @since 2.0.0
+     */
+    public static class PackageAnalyzer extends CodeAnalyzer {
+        @Override
+        public void init(CodeAnalysisContext analysisContext) {
+            analysisContext.addCompilationAnalysisTask(compilationAnalysisContext -> {
+                List<String> exportedModules = compilationAnalysisContext.currentPackage().manifest().exportedModules();
+                List<String> packageModules = new ArrayList<>();
+
+                // Add package modules to `packageModules` list
+                for (ModuleId moduleId : compilationAnalysisContext.currentPackage().moduleIds()) {
+                    Module module = compilationAnalysisContext.currentPackage().module(moduleId);
+                    packageModules.add(module.moduleName().toString());
+                }
+
+                for (String exportedModule : exportedModules) {
+                    if (!exportModuleExists(exportedModule, packageModules)) {
```

Comment:
we can use `packageModules.contains(exportedModule)` to make the code simpler without this private method

Reply:
Added